padle/padle.py

```diff
@@ -733,26 +736,26 @@ async def advance(self, ctx):
     @padleadmin.command()
     async def filter(self, ctx):
         """Re-creates the list of monsters"""
-        confirmation = await get_user_confirmation(ctx, "This command will take a long time to execute. Run anyways?")
-        if not confirmation:
-            return
         dbcog = await self.get_dbcog()
         mgraph = dbcog.database.graph
         final = []
-        # await ctx.send(mgraph.max_monster_id)
-        for i in range(1, 8800):  # hard coded bc max_monster_id as calculated above is 15k???
-            monster = await dbcog.find_monster(str(i), ctx.author)
-            with suppress(Exception):
-                nextTrans = mgraph.get_next_transform(monster)
-                prevTrans = mgraph.get_prev_transform(monster)
-                if (monster is not None and monster.name_en is not None and monster.on_na and
-                        monster.series.series_type != 'collab' and
-                        ((monster.sell_mp >= 50000 and (monster.superawakening_count > 1 or prevTrans is not None)) or
-                         "Super Reincarnated" in monster.name_en) and not monster.is_equip and
-                        nextTrans is None and monster.level >= 99):
-                    final.append(str(i))
+        for i in range(1, mgraph.max_monster_id):
+            monster = dbcog.get_monster(i)
+            if monster is None:
+                continue
+            if(self.is_valid_padle(monster, mgraph)):
+                final.append(str(i))
         with open("result.txt", "w") as file:
             file.write(",".join(final))
         with open("result.txt", "rb") as file:
             await ctx.send("List of monsters:", file=discord.File(file, "result.txt"))
         await ctx.tick()
+        
+    def is_valid_padle(monster, mgraph):
+        nextTrans = mgraph.get_next_transform(monster)
+        prevTrans = mgraph.get_prev_transform(monster)
+        return (monster.name_en is not None and monster.on_na and
+                monster.series.series_type != 'collab' and
+                ((monster.sell_mp >= 50000 and (monster.superawakening_count > 1 or prevTrans is not None)) or 
+                 "Super Reincarnated" in monster.name_en) and not monster.is_equip and
+                nextTrans is None and monster.level >= 99)
```

Comment:
This check is long. Is it possible to break it down into something easier to understand? Consider someone going in later to try to change it.

Here is an example where I group and comment them. It might be too much detail, but it is an example.
```py
(
    # Monster is in NA.
    monster.name_en is not None and monster.on_na and
    # Skip collab monsters.
    monster.series.series_type != 'collab' and
    # Only include Super Reincarnated monsters and advanced GFEs.
    (
        # GFEs with SAs or which have fully transformed.
        (monster.sell_mp >= 50000 and (monster.superawakening_count > 1 or prevTrans is not None)) or 
        # Or Super Reincarnated.
        "Super Reincarnated" in monster.name_en
    ) and
    # Skip equips.
    not monster.is_equip and
    # Skip transform bases.
    nextTrans is None and
    # Skip low-level monsters.
    monster.level >= 99
)
```

That middle block is weird to me. It doesn't include farmables (except Super Reincarnateds)? It skips Jiraiya (looped transform)? Is it supposed to look for endpoints of trees?